### DISCUSSION TITLE
Can Planetscale / Prisma use Fulltext search?

### DISCUSSION THREAD

**Discussion by chrisyeung1121** [prisma/prisma#11249] [Q&A]
## Problem

I understand since `3.8.0`, prisma has launch full text search support with MySQL.

Since I am using Planetscale, I am wondering if `referentialIntegrity` would affect the possibility to implement `fullTextSearch`?

**Comment by janpio** [comment]
Not that we are aware of. If you find out otherwise with a short test, let us know please and we will investigate.

**Comment by chrisyeung1121** [comment]
![ss1](https://user-images.githubusercontent.com/3488583/150107757-607a07a8-99bb-4761-a603-6a3593f68064.jpg)

I have received this error on `api/posts`

`posts.ts`

```
  if (req.method === 'GET') {
    // TODO: List all Post APIs
    const posts = await prisma.post.findMany({
      where: {
        content: {
          search: 'first'
        }
      }
    })
    res.json(posts)
  }
```

`schema.prisma`

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity", "fullTextSearch", "fullTextIndex"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("SHADOW_DATABASE_URL")
  referentialIntegrity = "prisma"
}

model Post {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  title     String   @db.VarChar(255)
  content   String   @db.Text

  @@fulltext([content])
}
```

**Reply by janpio** [reply]
Yep, looks broken. Can you turn this into a bug report issue so we can put it into our bug process? Thanks.

**Reply by robertf224** [reply]
ran into the same thing, noting as a work-around I was able to manually create the full-text index in planetscale, change my prisma schema to match it, and then full-text search worked fine

**Reply by janpio** [reply]
Can you please invest the time to create a separate issue about this @robertf224? We would love to reproduce and hopefully fix. Something has to be going wrong here.

**Comment by imMatheus** [comment]
Has this been resolved?

**Reply by janpio** [reply]
I am not aware there was ever an issue created as suggested in https://github.com/prisma/prisma/discussions/11249#discussioncomment-2190735 - so I don't think we ever had enough information to reproduce this.

**Reply by imMatheus** [reply]
The original question was asking if you could use full-text search with prisma and planetscale, and I don't know if that question has been answered.

**Reply by robertf224** [reply]
we didn't end up using PlanetScale (was just doing some exploration) so I never dug in further, all I recall per my comment was that I was able to get it working with the described work-around

**Reply by riccardogiorato** [reply]
I tried this with my planetscale project and it's fully working for me @imMatheus

**Reply by Olofu** [reply]
Please i can't get FULLTEXT search to work on planetscale via prisma. Vercel  always returns this error: "Cannot find a fulltext index to use for the search, try adding a @@fulltext([Fields...]) to your schema". I  have already added and enabled The previewFeatures included in the application's Prisma file.
 
It looks to me like the problem is my Prisma application isn't creating the necessary FULLTEXT indexes when I run my migrations or db push, which is why tried a workaround of creating them manually on planetscale. Why this isn't happening I don't know, but based on the other reports in the discussion it looks like a Prisma bug.

**Reply by janpio** [reply]
Did you trigger a deploy without cache at Vercel @Olofu? It might very well be that your deployed application just does not _know_ about the `@@fulltext`. (Best manually check if the index really, actually exists on the database level with a database GUI - I think Planetscale even has some web tooling that let's you check the current state of the database via the schema SQL)

**Reply by Olofu** [reply]
@janpio yes i did  trigger a deploy without cache at Vercel using VERCEL_FORCE_NO_BUILD_CACHE = 1 but still the issue persists. i have also checked manually on Planetscale and the index really, actually exists on the database.

**Reply by janpio** [reply]
The error message you are getting indicates that your Prisma Schema you are deploying does not include `@@fulltext`. Vercel has a button in the UI that you can use to redeploy an existing commit, without the cache. Do not check the `Redeploy with existing Build Cache.` box after selecting `Redeploy`.

**Reply by Olofu** [reply]
here is my schema.prisma 

generator client {
  provider = "prisma-client-js"
   previewFeatures = ["fullTextSearch", "fullTextIndex"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
  relationMode = "prisma"
}


model Buyer {
  proid          Int       @id @default(autoincrement())
  serialid       String
  displayid      String    @unique @default(uuid())
  state         String
  deal           String
  protype        String
  city           String
  area           String
  zone           String
  datetoday      DateTime  @default(now())
  authorId       String?
  author         User?     @relation(fields: [authorId], references: [userID], onDelete: Cascade)
  postsimage     Postimg[]

  @@index([authorId])
  @@fulltext([state])
  @@fulltext([city])
  @@fulltext([zone])
  @@fulltext([area])
  @@fulltext([protype])
  @@fulltext([deal])
}

every other thing regarding my database table works fine on Planetscale except the fulltext search

**Reply by janpio** [reply]
Please open a new discussion if we should debug your individual problem. Thanks.

**Reply by Olofu** [reply]
Thank you so much @janpio, i got it working, problem was from my end with the way i declared the fulltext index in my prisma schema i added @@fulltext([protype,deal,state,city]) because i intended to search multiple fields and it worked. Thanks to this article https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search

**Reply by janpio** [reply]
Ah, so the query used all 4 fields? Good you figured that out.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
